bot/cogs/moderation/infractions.py

```diff
@@ -225,7 +226,10 @@ def __init__(self, bot: Bot):
 
         self.mod_log.ignore(Event.member_remove, user.id)
 
-        action = user.kick(reason=reason)
+        if len(reason) > 512:
+            log.info("Kick reason is longer than 512 characters. Reason will be truncated for Audit Log.")
+
+        action = user.kick(reason=textwrap.shorten(reason, width=509, placeholder="..."))
```

Comment:
The `width` will ensure the new string will never be greater than the limit, **after appending placeholder** so you can change it to this

```suggestion
        action = user.kick(reason=textwrap.shorten(reason, width=512, placeholder=" ..."))
```

I suggest adding a space for `placeholder` as well.

You can do the same for `ban` and it'll look good to go.